src/runtime-rs/crates/hypervisor/src/qemu/network.rs

```diff
@@ -0,0 +1,372 @@
+// Copyright (c) 2024 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{convert::TryFrom, fs::File};
+
+use crate::{Address, NetworkConfig};
+use anyhow::{anyhow, Context, Result};
+
+#[derive(Debug, Default, PartialEq)]
+enum NetDeviceWithTransport {
+    // VirtioNetPCI is the virt-io pci networking device driver.
+    // "virtio-net-pci"
+    #[default]
+    VirtioNetPci,
+    // VirtioNetCCW is the virt-io ccw networking device driver.
+    // "virtio-net-ccw"
+    VirtioNetCcw,
+    //VhostUserNet represents a net vhostuser device type.
+    // "virtio-net"
+    VhostUserNet,
+    // VfioPCI is the vfio driver with PCI transport.
+    // "vfio-pci"
+    VfioPci,
+    // VfioCCW is the vfio driver with CCW transport.
+    // "vfio-ccw"
+    VfioCcw,
+    // VfioAP is the vfio driver with AP transport.
+    // "vfio-ap"
+    VfioAp,
+}
+
+impl ToString for NetDeviceWithTransport {
+    fn to_string(&self) -> String {
+        match self {
+            NetDeviceWithTransport::VirtioNetPci => "virtio-net-pci".to_owned(),
+            NetDeviceWithTransport::VirtioNetCcw => "virtio-net-ccw".to_owned(),
+            NetDeviceWithTransport::VhostUserNet => "virtio-net".to_owned(),
+            NetDeviceWithTransport::VfioPci => "vfio-pci".to_owned(),
+            NetDeviceWithTransport::VfioCcw => "vfio-ccw".to_owned(),
+            NetDeviceWithTransport::VfioAp => "vfio-ap".to_owned(),
+        }
+    }
+}
+
+impl TryFrom<&str> for NetDeviceWithTransport {
+    type Error = anyhow::Error;
+
+    // method to parse a PciPath from a string
+    fn try_from(s: &str) -> Result<Self> {
+        let transport = match s {
+            "virtio-net-pci" => NetDeviceWithTransport::VirtioNetPci,
+            "virtio-net-ccw" => NetDeviceWithTransport::VirtioNetCcw,
+            "virtio-net" => NetDeviceWithTransport::VhostUserNet,
+            "vfio-pci" => NetDeviceWithTransport::VfioPci,
+            "vfio-ccw" => NetDeviceWithTransport::VfioCcw,
+            "vfio-ap" => NetDeviceWithTransport::VfioAp,
+            _ => return Err(anyhow!("unsupported transport")),
+        };
+
+        Ok(transport)
+    }
+}
+
+#[derive(Debug, Default, PartialEq)]
+enum NetDeviceDriver {
+    #[default]
```

Comment:
I don't understand this enum. It is mixing guest visible entities (`VirtioNet`, `Vfio`) with host backends (`Vhostuser`)... What is it used for ?